### ISSUE TITLE
ReadyEvent not fired / fired at the wrong time

### ISSUE THREAD

**Issue by Teln0** [Discord4J/Discord4J#810]
**To Reproduce:**
```java
import discord4j.core.DiscordClientBuilder;
import discord4j.core.GatewayDiscordClient;
import discord4j.core.event.domain.lifecycle.ReadyEvent;
import discord4j.core.object.entity.User;

public class Bot {
  public static void main(String[] args) {
    GatewayDiscordClient client = DiscordClientBuilder.create("My token is here in the actual code").build().login().block();

    client.getEventDispatcher().on(ReadyEvent.class)
        .subscribe(event -> {
          User self = event.getSelf();
          System.out.println(String.format("Logged in as %s#%s", self.getUsername(), self.getDiscriminator()));
        });

    client.onDisconnect().block();
  }
}
```

**Expected Behavior:**
I'm expecting the discord tag of the bot to be printed.

**Actual Behavior:** 
The ReadyEvent is not fired.

**Version:** 
3.2.0

**Other:**

**Comment by danthonywalker** [comment]
Likely a race. Can you use `withEventDispatcher` in the gateway builder? `DiscordClient.create("token").gateway().withEventDispatcher()`

**Comment by quanticc** [comment]
This suggestion can work around this issue. A fix is planned while completing #796

I'll come back to this issue once that's cleared to verify the fix

**Comment by Teln0** [comment]
Ok, thank you ! I'll use 3.1.1 in the meantime

**Comment by quanticc** [comment]
This is ultimately an issue caused by reactor-core: https://github.com/reactor/reactor-core/issues/2513

Confirmed it is fixed in latest snapshots, so we'll wait until their 3.4.1 release to bump our 3.2.x version and revert the workaround (default event dispatcher change)

**Comment by quanticc** [comment]
Fixed via 581e099f7e5409da638438991b64335f6fe111c3
